pulsar4s-akka-streams/src/main/scala/com/sksamuel/pulsar4s/akka/streams/PulsarSourceGraphStage.scala

```diff
@@ -100,7 +109,7 @@ class PulsarSourceGraphStage[T](create: () => Consumer[T], seek: Option[MessageI
 
       override def shutdown()(implicit ec: ExecutionContext): Future[Done] = {
         completeStage()
-        consumer.closeAsync.map(_ => Done)
+        consumerOpt.fold(Future.successful(Done))(_.closeAsync.map(_ => Done))
```

Comment:
This is actually the main issue I noticed. If the consumer is not initialized yet, this will fail, and there are some situations where you might end up calling `shutdown()` during cleanup if the source fails.